netket/machine/torch.py

```diff
@@ -15,12 +21,15 @@ def _get_differentiable_parameters(m):
 
 
 class Torch(AbstractMachine):
-    def __init__(self, module, hilbert, dtype, outdtype=None):
+    def __init__(self, module, hilbert, dtype, outdtype=None, use_backpack=False):
         self._module = _torch.jit.load(module) if isinstance(module, str) else module
         self._module.double()
         self._n_par = _get_number_parameters(self._module)
         self._parameters = list(_get_differentiable_parameters(self._module))
-
+        self._use_backpack = use_backpack
+        if self._use_backpack:
+            assert "backpack" in sys.modules, "BackPACK was not successfully imported"
```

Comment:
The standard way to do such checks is to introduce a global variable, say `_BACKPACK_AVAILABLE` (caps because it's global and underscore because it's private) which you set to `True` if `import backpack` succeeds and to `False` otherwise. Then here you can simply ensure that `_BACKPACK_AVAILABLE` is true. Also this should definitely be a raise instead of an assert.

Reply:
The approach we have been following so far is to add a ```is_available``` variable in ```netket.utils```, for example you can see in there that we already have ```netket.utils.torch_available``` etc. So you could either add ```netket.utils.backpack_available``` for consistency with the other cases, or follow @twesterhout suggestion and localize it instead in the Torch file... Maybe I have a preference for the first solution, but it's not a big deal